2025.10: radius outpost crashes with EAP-TTLS

**Describe the bug**
The radius outpost crashes when authentication with EAP-TTLS

**To Reproduce**
Steps to reproduce the behavior:

1. Create `eapol_test` EAP-TTLS configuration (see https://github.com/BeryJu/radius-eap/issues/9)
2. Run eapol_test with EAP-TTLS configuration
3. See crash in the log

**Expected behavior**
EAP-TTLS should either work or state it is not (yet) supported if so but never crash.

**Logs**
```
authentik-outpost-radius-1  | {"code":"Access-Request","event":"Radius Request","id":117,"ip":"172.16.1.1","level":"info","logger":"authentik.outpost.radius","request":"3acacfba-938f-4406-946b-76db59db5e51","timestamp":"2025-10-28T10:45:04Z"}
authentik-outpost-radius-1  | {"code":"Access-Request","event":"Radius Request","id":0,"ip":"172.16.1.1","level":"info","logger":"authentik.outpost.radius","request":"dc5c4fa5-125b-4de9-96c3-caf6478c862f","timestamp":"2025-10-28T10:45:29Z"}
authentik-outpost-radius-1  | panic: runtime error: invalid memory address or nil pointer dereference
authentik-outpost-radius-1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x9ed471]
authentik-outpost-radius-1  | goauthentik.io/internal/outpost/radius.(*RadiusServer).Handle_AccessRequest(0xc0003d60b0, {0xd299c0, 0xc0000c61e0}, 0xc0002ffc98)
authentik-outpost-radius-1  | 	/go/src/goauthentik.io/internal/outpost/radius/handler.go:95 +0x1d4
authentik-outpost-radius-1  | goauthentik.io/internal/outpost/radius.(*RadiusServer).ServeRADIUS(0xc0003d60b0, {0xd299c0, 0xc0000c61e0}, 0xc0001d7080)
authentik-outpost-radius-1  | 	/go/src/goauthentik.io/internal/outpost/radius/handler.go:87 +0x98d
authentik-outpost-radius-1  | layeh.com/radius.(*PacketServer).Serve.func2({0xc00007c000, 0x80, 0x80}, {0xd2c220, 0xc000258c30})
authentik-outpost-radius-1  | 	/go/pkg/mod/layeh.com/radius@v0.0.0-20231213012653-1006025d24f8/server-packet.go:201 +0x573
authentik-outpost-radius-1 exited with code 0
```

**Version and Deployment (please complete the following information):**

-   authentik version: 2025.10.0
-   Deployment: docker-compose

I can reproduce this with no certificate selected in the provider:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x102d65038]

goroutine 98 [running]:
beryju.io/radius-eap.(*Packet).handleEAP(0x14000232190, {0x103097410, 0x1400019a180}, {0x1030930a0, 0x1400016a510}, 0x0)
	/Users/jens/.local/share/go/pkg/mod/beryju.io/radius-eap@v0.1.0/handler.go:98 +0xb8
beryju.io/radius-eap.(*Packet).HandleRadiusPacket(0x14000232190, {0x10308f320, 0x14000118660}, 0x1400020e300)
	/Users/jens/.local/share/go/pkg/mod/beryju.io/radius-eap@v0.1.0/handler.go:37 +0x1b0
goauthentik.io/internal/outpost/radius.(*RadiusServer).Handle_AccessRequest_EAP(0x14000442210, {0x10308f320, 0x14000118660}, 0x14000367c88)
	/Users/jens/dev/authentik/internal/outpost/radius/handler_eap.go:32 +0x260
goauthentik.io/internal/outpost/radius.(*RadiusServer).Handle_AccessRequest(0x14000442210, {0x10308f320, 0x14000118660}, 0x14000367c88)
	/Users/jens/dev/authentik/internal/outpost/radius/handler.go:95 +0x170
goauthentik.io/internal/outpost/radius.(*RadiusServer).ServeRADIUS(0x14000442210, {0x10308f320, 0x14000118660}, 0x1400020e300)
	/Users/jens/dev/authentik/internal/outpost/radius/handler.go:87 +0x718
layeh.com/radius.(*PacketServer).Serve.func2({0x14000212200, 0x80, 0x80}, {0x103091b20, 0x140000c0cf0})
	/Users/jens/.local/share/go/pkg/mod/layeh.com/radius@v0.0.0-20231213012653-1006025d24f8/server-packet.go:201 +0x47c
created by layeh.com/radius.(*PacketServer).Serve in goroutine 12
	/Users/jens/.local/share/go/pkg/mod/layeh.com/radius@v0.0.0-20231213012653-1006025d24f8/server-packet.go:146 +0x4b0
exit status 2
```

--- a/internal/outpost/radius/handler_eap.go
+++ b/internal/outpost/radius/handler_eap.go
@@ -41,95 +41,92 @@ func (pi *ProviderInstance) SetEAPState(key string, state *protocol.State) {
 }
 
 func (pi *ProviderInstance) GetEAPSettings() protocol.Settings {
-	protocols := []protocol.ProtocolConstructor{
-		identity.Protocol,
-		legacy_nak.Protocol,
+	settings := protocol.Settings{
+		Logger: &logrusAdapter{pi.log},
+		Protocols: []protocol.ProtocolConstructor{
+			identity.Protocol,
+			legacy_nak.Protocol,
+		},
 	}
 
 	certId := pi.certId
 	if certId == "" {
-		return protocol.Settings{
-			Protocols: protocols,
-		}
+		return settings
 	}
 
 	cert := pi.s.cryptoStore.Get(certId)
 	if cert == nil {
-		return protocol.Settings{
-			Protocols: protocols,
-		}
+		return settings
 	}
 
-	return protocol.Settings{
-		Logger:    &logrusAdapter{entry: pi.log},
-		Protocols: append(protocols, tls.Protocol, peap.Protocol),
-		ProtocolPriority: []protocol.Type{
-			identity.TypeIdentity,
-			tls.TypeTLS,
-		},
-		ProtocolSettings: map[protocol.Type]interface{}{
-			tls.TypeTLS: tls.Settings{
-				Config: &ttls.Config{
-					Certificates: []ttls.Certificate{*cert},
-					ClientAuth:   ttls.RequireAnyClientCert,
-				},
-				HandshakeSuccessful: func(ctx protocol.Context, certs []*x509.Certificate) protocol.Status {
-					ident := ctx.GetProtocolState(identity.TypeIdentity).(*identity.State).Identity
+	settings.Protocols = append(settings.Protocols, tls.Protocol, peap.Protocol)
+	settings.ProtocolPriority = []protocol.Type{
+		identity.TypeIdentity,
+		tls.TypeTLS,
+	}
+	settings.ProtocolSettings = map[protocol.Type]any{
+		tls.TypeTLS: tls.Settings{
+			Config: &ttls.Config{
+				Certificates: []ttls.Certificate{*cert},
+				ClientAuth:   ttls.RequireAnyClientCert,
+			},
+			HandshakeSuccessful: func(ctx protocol.Context, certs []*x509.Certificate) protocol.Status {
+				ident := ctx.GetProtocolState(identity.TypeIdentity).(*identity.State).Identity
 
-					ctx.Log().Debug("Starting authn flow")
-					pem := pem.EncodeToMemory(&pem.Block{
-						Type:  "CERTIFICATE",
-						Bytes: certs[0].Raw,
+				ctx.Log().Debug("Starting authn flow")
+				pem := pem.EncodeToMemory(&pem.Block{
+					Type:  "CERTIFICATE",
+					Bytes: certs[0].Raw,
+				})
+
+				fe := flow.NewFlowExecutor(context.Background(), pi.flowSlug, pi.s.ac.Client.GetConfig(), log.Fields{
+					"client":   utils.GetIP(ctx.Packet().RemoteAddr),
+					"identity": ident,
+				})
+				fe.Answers[flow.StageIdentification] = ident
+				fe.DelegateClientIP(utils.GetIP(ctx.Packet().RemoteAddr))
+				fe.Params.Add("goauthentik.io/outpost/radius", "true")
+				fe.AddHeader("X-Authentik-Outpost-Certificate", url.QueryEscape(string(pem)))
+
+				passed, err := fe.Execute()
+				if err != nil {
+					ctx.Log().Warn("failed to execute flow", "error", err)
+					return protocol.StatusError
+				}
+				ctx.Log().Debug("Finished flow")
+				if !passed {
+					return protocol.StatusError
+				}
+				access, _, err := fe.ApiClient().OutpostsApi.OutpostsRadiusAccessCheck(context.Background(), pi.providerId).AppSlug(pi.appSlug).Execute()
+				if err != nil {
+					ctx.Log().Warn("failed to check access: %v", err)
+					return protocol.StatusError
+				}
+				if !access.Access.Passing {
+					ctx.Log().Info("Access denied for user")
+					return protocol.StatusError
+				}
+				if access.HasAttributes() {
+					ctx.AddResponseModifier(func(r, q *radius.Packet) error {
+						rawData, err := base64.StdEncoding.DecodeString(access.GetAttributes())
+						if err != nil {
+							ctx.Log().Warn("failed to decode attributes from core: %v", err)
+							return errors.New("attribute_decode_failed")
+						}
+						p, err := radius.Parse(rawData, pi.SharedSecret)
+						if err != nil {
+							ctx.Log().Warn("failed to parse attributes from core: %v", err)
+							return errors.New("attribute_parse_failed")
+						}
+						for _, attr := range p.Attributes {
+							r.Add(attr.Type, attr.Attribute)
+						}
+						return nil
 					})
-
-					fe := flow.NewFlowExecutor(context.Background(), pi.flowSlug, pi.s.ac.Client.GetConfig(), log.Fields{
-						"client":   utils.GetIP(ctx.Packet().RemoteAddr),
-						"identity": ident,
-					})
-					fe.Answers[flow.StageIdentification] = ident
-					fe.DelegateClientIP(utils.GetIP(ctx.Packet().RemoteAddr))
-					fe.Params.Add("goauthentik.io/outpost/radius", "true")
-					fe.AddHeader("X-Authentik-Outpost-Certificate", url.QueryEscape(string(pem)))
-
-					passed, err := fe.Execute()
-					if err != nil {
-						ctx.Log().Warn("failed to execute flow", "error", err)
-						return protocol.StatusError
-					}
-					ctx.Log().Debug("Finished flow")
-					if !passed {
-						return protocol.StatusError
-					}
-					access, _, err := fe.ApiClient().OutpostsApi.OutpostsRadiusAccessCheck(context.Background(), pi.providerId).AppSlug(pi.appSlug).Execute()
-					if err != nil {
-						ctx.Log().Warn("failed to check access: %v", err)
-						return protocol.StatusError
-					}
-					if !access.Access.Passing {
-						ctx.Log().Info("Access denied for user")
-						return protocol.StatusError
-					}
-					if access.HasAttributes() {
-						ctx.AddResponseModifier(func(r, q *radius.Packet) error {
-							rawData, err := base64.StdEncoding.DecodeString(access.GetAttributes())
-							if err != nil {
-								ctx.Log().Warn("failed to decode attributes from core: %v", err)
-								return errors.New("attribute_decode_failed")
-							}
-							p, err := radius.Parse(rawData, pi.SharedSecret)
-							if err != nil {
-								ctx.Log().Warn("failed to parse attributes from core: %v", err)
-								return errors.New("attribute_parse_failed")
-							}
-							for _, attr := range p.Attributes {
-								r.Add(attr.Type, attr.Attribute)
-							}
-							return nil
-						})
-					}
-					return protocol.StatusSuccess
-				},
+				}
+				return protocol.StatusSuccess
 			},
 		},
 	}
+	return settings
 }

